src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs

```diff
@@ -572,45 +584,57 @@ private void setUnitAndDimension(ColumnMappingDTO model)
-                  setUnitAndDimension(model);
-                  break;
-               default:
-                  throw new NotImplementedException($"Setting description for unhandled column type: {model.CurrentColumnType}");
-            }
+            case null:
+               switch (model.CurrentColumnType)
+               {
+                  case ColumnMappingDTO.ColumnType.MetaData:
+                     model.Source = new MetaDataFormatParameter(model.ExcelColumn, model.MappingName, isColumn);
+                     break;
+                  case ColumnMappingDTO.ColumnType.Mapping:
+                     model.Source = new MappingDataFormatParameter(model.ExcelColumn,
+                        new Column() { Name = model.MappingName, Unit = new UnitDescription(UnitDescription.InvalidUnit) });
+                     setUnitAndDimension(model);
+                     break;
+                  default:
+                     throw new NotImplementedException($"Setting description for unhandled column type: {model.CurrentColumnType}");
+               }
 
-            _format.Parameters.Add(model.Source);
-         }
-         else if (model.Source is AddGroupByFormatParameter)
-         {
-            model.Source = new GroupByDataFormatParameter(model.ExcelColumn);
-            _format.Parameters.Add(model.Source);
-            setDataFormat(_format.Parameters);
-         }
-         else
-         {
-            model.Source.ColumnName = model.ExcelColumn;
-            if (model.CurrentColumnType == ColumnMappingDTO.ColumnType.MetaData)
+               _format.Parameters.Add(model.Source);
+               break;
+            case AddGroupByFormatParameter _:
+               model.Source = new GroupByDataFormatParameter(model.ExcelColumn);
+               _format.Parameters.Add(model.Source);
+               setDataFormat(_format.Parameters);
+               break;
+            default:
             {
-               (model.Source as MetaDataFormatParameter).IsColumn = isColumn;
+               model.Source.ColumnName = model.ExcelColumn;
+               switch (model.CurrentColumnType)
+               {
+                  case ColumnMappingDTO.ColumnType.MetaData:
+                     (model.Source as MetaDataFormatParameter).IsColumn = isColumn;
+                     break;
+                  case ColumnMappingDTO.ColumnType.Mapping:
+                     setUnitAndDimension(model);
+                     break;
+                  case ColumnMappingDTO.ColumnType.GroupBy:
+                     break;
+                  default:
+                     throw new NotImplementedException($"Setting description for unhandled column type: {model.CurrentColumnType}");
+               }
+
+               break;
             }
-            setUnitAndDimension(model);
          }
       }
 
```

Comment:
@msevestre here it is only this function SetDescriptionForRow that has been changed. The actual fix here is the moving setUnitAndDimension(model) call to the correct place. And I have also refactored a bit the switch/if...then..else statements  according to ReSharper suggestions.